### DISCUSSION TITLE
openapi_version staying in 3.1.0 does not matter the value

### DISCUSSION THREAD

**Discussion by Pierre-Alexandre35** [fastapi/fastapi#10748] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
def create_app() -> FastAPI:
    root_app = FastAPI(
        title="Backend-service",
        version="1.0.0",
        docs_url="/",
        openapi_version="3.0.0",
        openapi_url="/docs.json",
        redoc_url=None,
    )
```

### Description

So I want to use te openapi `3.0.0 `version because I am able to integreate it with Gitlab (unlike` 3.1.0`). However, i can put any value on my `openapi_version` variable, the swagger will always set the openapi version to `3.0.0`

```
{
    "openapi": "3.1.0",
    "info": {
        "title": "FastAPI",
        "version": "0.1.0"
    },
```

### Operating System

Linux

### Operating System Details

localhost

### FastAPI Version

0.103.1

### Pydantic Version

2.3.0

### Python Version

3.11

**Comment by waltermaldonado** [comment, marked as answer]
I am experiencing the same problem. If you take a look at the code, the `openapi_version` param is merely decorative, and will only be replaced in the final OpenAPI specs file, no matter the real version of the specs. 

This is being really problematic to me, since we have dependencies that rely on OpenAPI schemas of version 3.0.3, and changing this version will not prevent the generation of 3.1.0 specs, with the `type: "null"` specification, for example...

This is actually _everything_ the `openapi_version` param is used for (line 465)
https://github.com/tiangolo/fastapi/blob/6761fc1fa4ffce2cc0c73117aeadce81fa3a659c/fastapi/openapi/utils.py#L465):
![image](https://github.com/tiangolo/fastapi/assets/8573950/d241bac1-1092-44e3-a3a1-56c5fdc3145a)
